compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/array-access-assignment.expect.md

```diff
@@ -30,30 +30,21 @@ export const FIXTURE_ENTRYPOINT = {
 ```javascript
 import { c as _c } from "react/compiler-runtime";
 function Component(t0) {
-  const $ = _c(6);
-  const { a, b, c } = t0;
+  const $ = _c(2);
   let t1;
-  if ($[0] !== a || $[1] !== b || $[2] !== c) {
+  if ($[0] !== t0) {
+    const { a, b, c } = t0;
```

Comment:
this loses memoization precision, need to debug why the mutable range ends up spanning this instruction